lib/requester/util.js

```diff
@@ -68,14 +68,23 @@ var _ = require('lodash'),
      * @param lookupOptions
      * @param lookupOptions.port
      * @param lookupOptions.restrictedAddresses
+     * @param {Object.<String, String | {ip: String, family: Number}>} lookupOptions.hostIpMap An object containing host
+     * to ip mapping. The value is either a string (family defaults to 4 in this case) or an object having ip and family
      * @param hostname
      * @param options
      * @param callback
      */
     lookup = function (lookupOptions, hostname, options, callback) {
-        var self = this;
+        var self = this,
+            lowercaseHost = hostname.toLowerCase(),
+            resolvedAddr;
+
+        if (lowercaseHost !== LOCALHOST) {
+            if (lookupOptions.hostIpMap && (resolvedAddr = lookupOptions.hostIpMap[lowercaseHost])) {
+                typeof resolvedAddr === 'string' && (resolvedAddr = {ip: resolvedAddr, family: 4});
+                return callback(null, resolvedAddr.ip, resolvedAddr.family || 4);
```

Comment:
We blindly trust this to be an IP address? What if it is not?

Reply:
yes, will add checks for it.
Also, I just realized that this is bypassing the check for `restrictedAddresses`. Will fix that too.